src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/RuntimeInformation.Unix.cs

```diff
@@ -11,8 +11,8 @@ public static partial class RuntimeInformation
         private static readonly object s_processLock = new object();
         private static string? s_osPlatformName;
         private static string? s_osDescription;
-        private static Architecture? s_osArch;
-        private static Architecture? s_processArch;
+        private static int s_osArch = -1;
```

Comment:
I don't think we need the locks at all, but if you keep them, how about keeping just one?